core/src/main/scala/aima/core/environment/vacuum/SimpleReflexVacuumAgentProgram.scala

```diff
@@ -13,6 +13,5 @@ class SimpleReflexVacuumAgentProgram extends SimpleReflexAgentProgram[VacuumPerc
     case LocationAPercept => RightMoveAction
     case LocationBPercept => LeftMoveAction
     case CleanPercept     => NoAction
-    case NoPercept        => NoAction
```

Comment:
I'm not sure I like changing from NoPercept and NoAction to Option[Percept] and Option[Action] .
While it is common to represent optionality  with Option I find it harder to do things like fold.
Now I have to do map and fold and get or else whereas before I could just do fold easily.